File: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/dominio/indices/IndiceSemCorrecao.java
/*
 * Decompiled with CFR 0.152.
 */
package br.jus.trt8.pjecalc.negocio.dominio.indices;

import br.jus.trt8.pjecalc.base.comum.HelperDate;
import br.jus.trt8.pjecalc.base.comum.Periodo;
import br.jus.trt8.pjecalc.negocio.comum.rotinasdecalculo.CalculadorDeIndices;
import br.jus.trt8.pjecalc.negocio.dominio.indices.IndiceBase;
import br.jus.trt8.pjecalc.negocio.dominio.indices.api.IndiceDeCalculo;
import br.jus.trt8.pjecalc.negocio.dominio.indices.tr.RepositorioDeIndiceTR;
import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

public class IndiceSemCorrecao
extends IndiceBase {
    private static final long serialVersionUID = 1L;

    public IndiceSemCorrecao(Date competencia) {
        super(RepositorioDeIndiceTR.class, competencia, BigDecimal.ZERO);
        this.setValorAcumulado(BigDecimal.ONE);
    }

    public static List<IndiceDeCalculo> obterTabela(Periodo periodo) {
        ArrayList<IndiceSemCorrecao> lista = new ArrayList<IndiceSemCorrecao>();
        for (Periodo competencia : HelperDate.breakInMonths(periodo.getInicial(), periodo.getFinal())) {
            lista.add(new IndiceSemCorrecao(HelperDate.getCurrentCompetence(competencia.getInicial()).getDate()));
        }
        return CalculadorDeIndices.calcularIndiceAcumulado(lista);
    }

    @Override
    public IndiceBase validar() {
        return this;
    }

    @Override
    public IndiceBase validarParaConsulta() {
        return this;
    }

    @Override
    public boolean existe() {
        return false;
    }

    @Override
    public Object obterChavePrimaria() {
        return null;
    }

    @Override
    public IndiceBase clonar() {
        return this;
    }
}

File: src/lib/pjecalc/core/dominio/indices/indice-sem-correcao.ts
/**
 * PJe-Calc v2.15.1 — IndiceSemCorrecao
 * Porte 1:1 de: br.jus.trt8.pjecalc.negocio.dominio.indices.IndiceSemCorrecao
 *
 * Ref Java: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/dominio/indices/IndiceSemCorrecao.java
 *
 * Índice "sem correção": valorAcumulado = 1 para toda competência.
 * Usado quando `IndiceMonetarioEnum.SEM_CORRECAO` é selecionado.
 */
import Decimal from 'decimal.js';
import { IndiceBase } from './indice-base';
import { HelperDate } from '../../base/comum/helper-date';
import type { Periodo } from '../../base/comum/periodo';
import { calcularIndiceAcumulado } from '../../comum/rotinasdecalculo/calculador-de-indices';

export class IndiceSemCorrecao extends IndiceBase {
  constructor(competencia: Date) {
    super(competencia, new Decimal(0));
    this.setValorAcumulado(new Decimal(1));
  }

  /** obterTabela (linha 26) — breakInMonths + IndiceSemCorrecao por mês */
  static obterTabela(periodo: Periodo): IndiceSemCorrecao[] {
    const lista: IndiceSemCorrecao[] = [];
    for (const mes of HelperDate.breakInMonths(periodo.getInicial(), periodo.getFinal())) {
      const comp = HelperDate.getCurrentCompetence(mes.getInicial()).getDate();
      lista.push(new IndiceSemCorrecao(comp));
    }
    return calcularIndiceAcumulado(lista) as IndiceSemCorrecao[];
  }

  clonar(): IndiceSemCorrecao {
    return new IndiceSemCorrecao(this.getCompetencia());
  }
}
